File: java/logic2/makeChocolate.java
package java.logic2;

public class makeChocolate {

  public int makeChocolate(int small, int big, int goal) {
    int weight = 0;

    for (int i = 0; i < big; i++) {
      weight += 5;
    }

    while (weight > goal) {
      weight -= 5;
    }

    if (weight == goal)
      return 0;

    for (int i = 1; i <= small; i++) {
      weight++;
      if (weight == goal)
        return i;
    }

    return -1;

  }

}

File: python/logic2/make_chocolate.py
def makeChocolate(small, big, goal):
    weight = 0

    for i in range(big):
        weight += 5

    while (weight > goal):
        weight -= 5

    if (weight == goal): return 0

    for j in range(small + 1):
        weight += 1
        if (weight == goal): return j

    return -1


print(makeChocolate(6, 2, 7))